lxd/container_lxc.go

```diff
@@ -6883,13 +6887,155 @@ func (c *containerLXC) insertProxyDevice(devName string, m types.Device) error {
 	return nil
 }
 
+func (c *containerLXC) doNat(proxy string, device types.Device) error {
+	// Check NAT requirements
+	if device["bind"] != "host" {
+		return errors.New("NAT requirements not met")
+	}
+
+	listenAddr, err := parseAddr(device["listen"])
+	if err != nil {
+		return err
+	}
+
+	connectAddr, err := parseAddr(device["connect"])
+	if err != nil {
+		return err
+	}
+
+	address, _, err := net.SplitHostPort(connectAddr.addr[0])
+	if err != nil {
+		return err
+	}
+
+	if len(connectAddr.addr) > len(listenAddr.addr) {
+		// Cannot support single port -> multiple port
+		return errors.New("Cannot map single port to multiple ports")
+	}
+
+	// Support TCP <-> TCP and UDP <-> UDP
+	if listenAddr.connType == "unix" || connectAddr.connType == "unix" ||
+		listenAddr.connType != connectAddr.connType {
+		return fmt.Errorf("%s <-> %s not supported", listenAddr.connType,
+			connectAddr.connType)
+	}
```

Comment:
Another check that feels better done at config validation time rather than this late.